File: loan_management_system/models/Payment.js
const mongoose = require('mongoose');

const PaymentSchema = new mongoose.Schema({
  loan: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Loan',
    required: true,
  },
  dueDate: {
    type: Date,
    required: true,
  },
  amount: {
    type: Number,
    required: true,
  },
  status: {
    type: String,
    enum: ['pending', 'paid'],
    default: 'pending',
  },
  paymentDate: {
    type: Date,
  },
});

module.exports = mongoose.model('Payment', PaymentSchema);

File: loan_management_system/android_app/app/src/main/java/com/example/loanmanagement/data/model/Payment.kt
package com.example.loanmanagement.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

@Entity(
    tableName = "payments",
    foreignKeys = [
        ForeignKey(
            entity = Loan::class,
            parentColumns = ["id"],
            childColumns = ["loanId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Payment(
    @PrimaryKey
    @SerializedName("_id")
    val id: String,

    @SerializedName("loan")
    val loanId: String,

    @SerializedName("dueDate")
    val dueDate: String,

    @SerializedName("amount")
    val amount: Double,

    @SerializedName("status")
    val status: String, // "pending" or "paid"

    @SerializedName("paymentDate")
    val paymentDate: String?
)
